ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -147,10 +147,18 @@ layout (set = 0, binding = 5, row_major, std430) readonly restrict buffer InstDa
 	InstanceData data[];
 } InstData;
 
-vec3 irr_glsl_MC_getCamPos()
+vec3 irr_glsl_MC_getNormalizedWorldSpaceV()
 {
 	vec3 campos = irr_glsl_SBasicViewParameters_GetEyePos(CamData.params.NormalMatAndEyePos);
-	return campos;
+	return normalize(campos - WorldPos);
+}
+vec3 irr_glsl_MC_getNormalizedWorldSpaceN()
+{
+	return normalize(Normal);
+}
+vec3 irr_glsl_MC_getWorldSpacePosition()
+{
+	return WorldPos;
```

Comment:
what do you need worldspace position for?

Reply:
ah ok, its the world space pos of the frament... I misread and thought it was the camera.